lib/datadog/appsec/configuration/settings.rb

```diff
@@ -116,9 +116,14 @@ def duration(base = :ns, type = :integer)
 
         def initialize
           @integrations = []
+          # Stores which options have been configured using Datadog.configure block or ENV variables
+          @configured = Set.new
```

Comment:
Minor: Remember to `require 'set'` -- it's not automatically available on all Ruby versions (so currently this code is relying on something else doing that require).